CONTRIBUTING.md

```diff
@@ -1,16 +1,256 @@
+
+* RIOT provides static test tools to verify the quality of changes (flake8,
+  trailing whitespaces, documentation, etc). These tools are wrapped in a
+  single `make` target: `static-test`.
+  *watch out:* the command below will rebase your branch on your master branch,
+  so make sure they can be rebased (e.g. there's no potential conflict).
+
+      $ make static-test
+
+  Use it before opening a PR to perform last time checks.
+
+* Each commit should target changes of specific parts/modules of RIOT. The
+  commits use the following pattern:
+  `path/to/module/directory: description of changes`.
+  You can use multi-line commit messages if you want to detail more the
+  changes.
+
+* Try to answer reviews as quickly as possible to speed up the review process
+  and avoid stalled PRs.
+
+You can find more information about RIOT development procedure on this
+[wiki page][development-procedures].
+
+[existing-pull-requests]: https://github.com/RIOT-OS/RIOT/pulls
+[coding-conventions]: https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions
+[uncrustify]: http://uncrustify.sourceforge.net
+[development-procedures]: https://github.com/RIOT-OS/RIOT/wiki/Development-procedures
+
+### Git usage
+
+Using git is a bit difficult for newcomers. If you are completely new to git,
+we recommend that you [start by learning it](https://try.github.io/) a bit.
+
+In this section, we give the bare minimum for a better experience with our
+development workflow on GitHub.
+
+[git-scm-getting-started]: https://git-scm.com/book/en/v2/Getting-Started-Git-Basics
+
+#### Setup your local RIOT repository
+
+Before you start modifying code, you need to fork the RIOT upstream repository
+from the [RIOT main github page][riot-github].
+
+If it's your first time with git, configure your name and emails:
+
+    $ git config --global user.name = "<your name here>"
+    $ git config --global user.email = "<your email address here>"
+
+Then clone locally your fork of RIOT (replace `account name` with your actual
+login on GitHub):
+
+    $ git clone git@github.com:<account name>/RIOT.git
+
+You can keep any branch of your local repository up-to-date with the upstream
+master branch with the following commands:
+
+    $ git checkout <branch name>
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+
+Use it at least before opening a PR. This will ensure the PR can be merged.
+
+[riot-github]: https://github.com/RIOT-OS/RIOT
+
+#### Work on branches
+
+Avoid opening PR from the `master` branch of your fork to the master branch of
+the RIOT upstream repository: update your master branch and start a new branch
+from it.
+
+    $ git checkout master
+    $ git pull --rebase https://github.com/RIOT-OS/RIOT.git
+    $ git checkout -b <new branch>
+    # Do your changes, commit, update with latest upstream master
+    $ git push
+
+This is important because it will avoid the maintainer to have to deal with
+his master branch yours if he wants to test your changes locally.
```

Comment:
This sentence needs some word-smithing ;-).

Also, the names of a local branch and a remote tracking branch are independent, so this sentence doesn't make much sense in general. A local branch "that-weird-pr-by-that-gal" can track "mueller-meyer-schmidt/master" and you don't even have to create a new branch to check out their contribution:

```sh
git fetch mueller-meyer-schmidt
git checkout mueller-meyer-schmidt master
```

if a PR exists of the contribution you don't even have to fetch their remote (my preferred method for testing PRs):

```sh
git fetch https://github.com/RIOT-OS/RIOT.git refs/pull/${PR}/head
git checkout FETCH_HEAD
```

The reason why a separate branch is preferred over master is because it is best practice to keep changes apart from mainline (aka master) and from each other (if you have multiple contributions). But IMHO we should not force contributors into a schema for their repository, based on false claims.